samples/test/deletes.js

```diff
@@ -0,0 +1,230 @@
+              timestamp: TIMESTAMP,
+            },
+            connected_wifi: {
+              value: 1,
+              labels: [],
+              timestamp: TIMESTAMP,
+            },
+            os_build: {
+              value: 'PQ2A.190405.004',
+              labels: [],
+              timestamp: TIMESTAMP,
+            },
+          },
+          cell_plan: {
+            data_plan_05gb: {
+              value: true,
+              labels: [],
+              timestamp: TIMESTAMP,
+            },
+          },
+        },
+      },
+      {
+        key: 'phone#4c410523#20190505',
+        data: {
+          stats_summary: {
+            connected_cell: {
+              value: 0,
+              labels: [],
+              timestamp: TIMESTAMP,
+            },
+            connected_wifi: {
+              value: 1,
+              labels: [],
+              timestamp: TIMESTAMP,
+            },
+            os_build: {
+              value: 'PQ2A.190406.000',
+              labels: [],
+              timestamp: TIMESTAMP,
+            },
+          },
+          cell_plan: {
+            data_plan_05gb: {
+              value: true,
+              labels: [],
+              timestamp: TIMESTAMP,
+            },
+          },
+        },
+      },
+      {
+        key: 'phone#5c10102#20190501',
+        data: {
+          stats_summary: {
+            connected_cell: {
+              value: 1,
+              labels: [],
+              timestamp: TIMESTAMP,
+            },
+            connected_wifi: {
+              value: 1,
+              labels: [],
+              timestamp: TIMESTAMP,
+            },
+            os_build: {
+              value: 'PQ2A.190401.002',
+              labels: [],
+              timestamp: TIMESTAMP,
+            },
+          },
+          cell_plan: {
+            data_plan_10gb: {
+              value: true,
+              labels: [],
+              timestamp: TIMESTAMP,
+            },
+          },
+        },
+      },
+      {
+        key: 'phone#5c10102#20190502',
+        data: {
+          stats_summary: {
+            connected_cell: {
+              value: 1,
+              labels: [],
+              timestamp: TIMESTAMP,
+            },
+            connected_wifi: {
+              value: 0,
+              labels: [],
+              timestamp: TIMESTAMP,
+            },
+            os_build: {
+              value: 'PQ2A.190406.000',
+              labels: [],
+              timestamp: TIMESTAMP,
+            },
+          },
+          cell_plan: {
+            data_plan_10gb: {
+              value: true,
+              labels: [],
+              timestamp: TIMESTAMP,
+            },
+          },
+        },
+      },
+    ];
+
+    await table.insert(rowsToInsert);
+  });
+
+  function runAndSnapshot(cmd) {
+    const deleteSnippetsCommand = `node deleteSnippets ${INSTANCE_ID} ${TABLE_ID} ${cmd}`;
+    const stdout = execSync(deleteSnippetsCommand);
+    snapshot(stdout);
```

Comment:
Snapshot is a special library that the other sample tests use which does two things:
1. If there is no snapshot yet for this test case it produces a snapshot
2. If there _is_ a snapshot then it will compare this snapshot with `stdout` and throw an assertion error if they are not the same.

So assertions are built into snapshot. This is actually a really efficient way to generate test cases and make concise comparisons without writing too much code. It is used in `samples/test/filters.js`.